src/network/MissionControlProtocol.cpp

```diff
@@ -129,25 +130,29 @@ void MissionControlProtocol::handleDriveRequest(const json& j) {
 	this->setRequestedCmdVel(dtheta, dx);
 }
 
-void MissionControlProtocol::handleSetArmIKEnabled(const json& j) {
+void MissionControlProtocol::handleRequestArmIKEnabled(const json& j) {
 	bool enabled = j["enabled"];
 	if (enabled) {
-		Globals::armIKEnabled = false;
-		if (_arm_ik_repeat_thread.joinable()) {
-			_arm_ik_repeat_thread.join();
+		if (!Globals::armIKEnabled) {
+			if (_arm_ik_repeat_thread.joinable()) {
+				_arm_ik_repeat_thread.join();
+			}
+			DataPoint<navtypes::Vectord<Constants::arm::IK_MOTORS.size()>> armJointPositions =
+				robot::getMotorPositionsRad(Constants::arm::IK_MOTORS);
+			// Rover responds with armIKEnabledReport after requestArmIKEnable is processed
+			if (armJointPositions.isValid()) {
+				Globals::planarArmController.set_setpoint(armJointPositions.getData());
+				this->setArmIKEnabled(true);
+				_arm_ik_repeat_thread =
+					std::thread(&MissionControlProtocol::updateArmIKRepeatTask, this);
+			} else {
+				// unable to enable IK
+				LOG_F(WARNING, "Unable to enable IK");
+				this->sendArmIKEnabledReport();
```

Comment:
I know `Globals::armIKEnabled` is already false, but could we change this to call `setArmIKEnabled(false)`? Otherwise, it looks weird that the `if` clause calls it but the `else` clause doesn't, since it's not immediately obvious what the difference between `setArmIKEnabled` and `sendArmIKEnabledReport` are.